batchflow/models/torch/base.py

```diff
@@ -1720,25 +1727,26 @@ def save(self, path, use_onnx=False, path_onnx=None, use_openvino=False, path_op
             ignore_attributes = []
         ignore_attributes = set(ignore_attributes)
 
-        if use_onnx:
+        if format == "onnx":
             if batch_size is None:
                 raise ValueError('Specify valid `batch_size`, used for model inference!')
 
             inputs = self.make_placeholder_data(batch_size=batch_size, unwrap=False)
-            path_onnx = path_onnx or (path + '_onnx')
+            path_onnx = path if path.endswith(".onnx") else os.path.join(dirname, "model.onnx")
```

Comment:
There is no way to call `model.save('my_model.abcd', format="onnx", pickle_metadata=False)` and get a model by path `'my_model.abcd'` which user specified. User must be able to choose extension which he want.

So 
1) use `path` as `path_onnx` if `pickle_metadata=False`,
2) use `os.path.splitext(path)[0]` + '.onnx` if `pickle_metadata=True`.

In that logic `path` is a path to the main file which is dumped. If `pickle_metadata=False`, it is the model itself, if `pickle_metadata=True`, it is pickled file.

The same for other formats.